src/ThreadPool/ThreadPool.h

```diff
@@ -0,0 +1,135 @@
+#pragma once
+
+#include "StructParameters.h"
+#include "StructQueue.h"
+
+#include <arpa/inet.h>
+
+#include "../Factory/MapCreator.h"
+
+#include <atomic>
+#include <mutex>
+#include <queue>
+#include <thread>
+#include <vector>
+
+namespace MapCreator {
+class MapCreator;
+}
+
+namespace ThreadPool {
+
+class Queue {
+public:
+  /**
+   * @brief Construct a new Queue object
+   *
+   * @param sockfd the socket file descriptor
+   * @param connectAddress the connection address - used when doing accept
+   * @param numberOfThreads number of threads to support the operation.
+   *                        if it equals to 1, then the server is serial.
+   */
+  explicit Queue(const int sockfd, const sockaddr_in connectAddress,
```

Comment:
C,417,Code is not generic
C,418,Breaking encapsulation (I'm not looking at this class anymore, the design should be vastly improved!)